tap_playvox/sync.py

```diff
@@ -24,52 +23,52 @@ def sync_endpoint(client,
                   endpoint,
                   key_bag,
                   stream_params={}):
-    
+
     persist = endpoint.get('persist', True)
 
     if persist:
         stream = catalog.get_stream(stream_name)
         schema = stream.schema.to_dict()
         mdata = metadata.to_map(stream.metadata)
         write_schema(stream)
-    
+
     # API Parameters
     # Reference https://support.agyletime.com/hc/en-us/articles/4402740869401-Integration-Guide-Agent-Metrics
     initial_load = True
     next_page_token=''
-  
+
     params={}
-    
+
     #ISO 8601 Datetime format
     iso_format = "%Y-%m-%dT%H:%M:%S.%fZ"
     start_date = singer.get_bookmark(state,
                                      stream_name,
-                                     client.start_date)
+                                     key='endDate',
+                                     default=client.start_date)
+
+    start_datetime = singer.utils.strptime_to_utc(start_date)
+
+    # Subtract 14 days from the bookmark datetime for Schedule Metrics since there can be updates
+    # during this period
+    if stream_name == 'schedule_metrics':
+        start_datetime -= timedelta(days=14)
+
+    iso_start_datetime = start_datetime.strftime(iso_format)
 
-    if start_date:
-        start_datetime = singer.utils.strptime_to_utc(start_date)
-    else:
-        # If no start_date or bookmark available, default to start_date of the config
-        start_datetime = singer.utils.strptime_to_utc(client.start_date)
-        
-        dt_string = str(start_datetime)
-        dt_object = datetime.fromisoformat(dt_string)
-        start_datetime = dt_object.strftime(iso_format)
-    
     if stream_name == 'tasks':
         params = {
-            'taskStartTimeFrom': str(start_datetime),
+            'taskStartTimeFrom': iso_start_datetime,
             'taskStartTimeTo': str(datetime.utcnow().isoformat())
         }
- 
-    if stream_name == 'agent_metrics':
+
+    if stream_name == 'agent_metrics' or stream_name == 'schedule_metrics':
```

Comment:
there's no such stream called schedule_metrics in the ENDPOINT_CONFIG, so none of the branches that check this run. https://github.com/Pathlight/tap-playvox/blob/39ae092845c709d624e363beaab0f83f3ebbdd81/tap_playvox/endpoints.py#L1